examples/user_guide/Dependencies_and_Watchers.ipynb

```diff
@@ -0,0 +1,820 @@
+    "- [Event Parameter](#Event-Parameter): Special Parameter type providing triggerable transient Events (like a momentary push button)\n",
+    "- [Async executor](#Async-executor): Support for asynchronous processing of Events, e.g. for interfacing to external servers\n",
+    "\n",
+    "Each of these will be described in the following sections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86823025",
+   "metadata": {},
+   "source": [
+    "### `batch_call_watchers`\n",
+    "\n",
+    "Context manager that accumulates parameter changes on the supplied object and dispatches them all at once when the context is exited, to allow multiple changes to a given parameter to be accumulated and short-circuited, rather than prompting serial changes from a batch of parameter setting:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "898288b3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with param.parameterized.batch_call_watchers(p):\n",
+    "    p.a = 2\n",
+    "    p.a = 3\n",
+    "    p.a = 1\n",
+    "    p.a = 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ecb4e66b",
+   "metadata": {},
+   "source": [
+    "Here, even though `p.a` is changed four times, each of the watchers of `a` is executed only once, with the final value. One of those events then changes `b`, so `b`'s watcher is also executed once.\n",
+    "\n",
+    "If we set `b` explicitly, `b`'s watcher will be invoked twice, once for the explicit setting of `b`, and once because of the code `self.b += 1`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c8b126be",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with param.parameterized.batch_call_watchers(p):\n",
+    "    p.a = 2\n",
+    "    p.b = 8\n",
+    "    p.a = 3\n",
+    "    p.a = 1\n",
+    "    p.a = 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54868160",
+   "metadata": {},
+   "source": [
+    "###  `discard_events`\n",
+    "\n",
+    "Context manager that discards any events within its scope that are triggered on the supplied parameterized object. Useful for making silent changes to dependent parameters, e.g. in a setup phase. If your dependencies are meant to ensure consistency between parameters, be careful that your manual changes in this context don't put the object into an inconsistent state!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bd655822",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with param.parameterized.discard_events(p):\n",
+    "    p.a = 2\n",
+    "    p.b = 9"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "04bd8a67",
+   "metadata": {},
+   "source": [
+    "(Notice that none of the callbacks is invoked, despite all the Watchers on `p.a` and `p.b`.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dd987890",
+   "metadata": {},
+   "source": [
+    "### `.param.trigger`\n",
```

Comment:
Would be nice to mention that when modifying mutable parameters no events are triggered and that `.param.trigger` is frequently useful in such cases.

Reply:
Isn't that covered by the following paragraph already? Does it need more explanation?

Reply:
Yes, sorry did not read in sufficient detail.